### PR TITLE
Consider all running clocks when doing clock sync.

### LINKED ISSUE #191: Clocks not synced if manually started/stopped
Maimed reported at the European Cup that the clocks got out of sync. Current theory is that this was operator error, as it corrected itself at the next jam. Thus it seems that manually starting/stopping clocks may get things out of sync. Not the end of the world (or something operators should have to do), but probably worth looking into.

### PR DESCRIPTION
**Pull request by brian-brazil**
This is simpler, and avoids going out of sync.

Fix precision issue in isDisplayChange.

Fixes #191

### CODE REVIEW

File: src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java [hunk 1]
```diff
@@ -363,18 +359,13 @@ public UpdateClockTimerTask() {
...
-                    // with respect to the master clock
-                    long nowMs = currentTime % 1000;
-                    if (masterClock != null) {
-                        nowMs = masterClock.getTime() % 1000;
-                        if (masterClock.isCountDirectionDown()) {
-                            nowMs = (1000 - nowMs) % 1000;
-                        }
+                    // with respect to the running clocks.
+                    ClockImpl masterClock = clocks.get(0);
+                    long nowMs = masterClock.getTime() % 1000;
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java:366]: This is hiding a class variable.

**Review by frank-weinberg — COMMENTED** [review]
This seems to sporadically fail the clockSync test. (Seen it once  in 5-10 runs.)

File: src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java (hunk 1 above)
Comment (brian-brazil) [reply src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java:366]: Ah, forgot to remove that.

Let me dig into that test. Things were test-order dependent previously, but this change should have fixed that in passing....

Comment (brian-brazil) [reply src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java:366]: I've run all the tests for ~2.5 hours, with no failures.

**Pull request merged**